src/widgets/panel/panel.tsx

```diff
@@ -76,18 +76,30 @@ export type PanelTemplates = {
   collapseButtonText?: Template<{ collapsed: boolean }>;
 };
 
-export type PanelWidgetOptions = {
+export type PanelWidgetOptions<TWidget extends WidgetFactory<any, any, any>> = {
```

Comment:
not sure if this is the user api, but I think the widget should be optional, and we can do something like `extends WF | never = never` and then do a conditional on line 86 where if it's given we do the current code and if it extends never we do either an object or the union of all widget render states. What do you think?

Reply:
I just played with the idea. The issue is, when TWidget is not given, it's either

1. user is using a custom widget
2. user is using one of the built-in widgets but just not providing the type.

Let's say it's 2 and a breadcrumb. Showing the union of all widget render states might help users, but also can lead them to a wrong type, because it will also show things for searchBox, autocomplete, ... it might be confusing.

And if it's 1, showing the union really doesn't help at all.

The current implementation, when no generic is given, will just type widgetRenderState as `any`. They didn't provide the type, so we're giving them back `any`. Sort of makes sense? 
What do you think?

Reply:
ah yeah that does make sense. Might be better even to type it with unknown :) I don't know if we can find a place to document when and why to add the generic though

Reply:
![image](https://user-images.githubusercontent.com/499898/97875520-ca943d80-1d1a-11eb-9087-2f79b71b265a.png)

https://github.com/algolia/instantsearch.js/pull/4527/commits/0fcc02dede5938c926c6aec766a904ef6df3be57
trying to understand this error message, but cannot.
Can you help me?

Reply:
unknown is a specific type, and the breadcrumb doesn't fit in unknown, but it does in any (as far as I can tell)

Reply:
In my understanding, 

```ts
type Aa = unknown;
const a: Aa = '3'
```

this should work and it does [here](https://www.typescriptlang.org/play?#code/C4TwDgpgBAgghlAvFArgOwNZoPYHc0DcAUAMbZoDOwUcAXLAsgOQDMTRHQA).

> Anything is assignable to unknown

https://www.typescriptlang.org/docs/handbook/release-notes/typescript-3-0.html#new-unknown-top-type

but the error message in the screenshot doesn't say something cannot be assigned to `unknown`, but the other way around.

Do you think we can revert it back to `any`, so that it won't harm the user-land typing if we're not going to deal with this now?